src/components/Dashboard/FilingHistoryList.vue

```diff
@@ -374,1042 +376,665 @@
-            // clear spinner on error
-            this.$root.$emit('showSpinner', false)
+  /** Loads the comments and documents for this history item. */
+  private async loadCommentsAndDocuments (item: HistoryItemIF): Promise<void> {
+    // fetch comments and documents from API
+    const data = await Promise.all([
+      this.fetchComments(item.commentsLink),
+      this.fetchDocuments(item.documentsLink)
+    ]).catch(error => {
+      // eslint-disable-next-line no-console
+      console.log('fetchComments() or fetchDocuments() error =', error)
+      // FUTURE: enable some error dialog?
+    })
+
+    if (data && data.length === 2) {
+      // load the comments
+      item.comments = this.flattenAndSortComments(data[0])
+      // load the documents
+      item.documents = data[1] || []
+
+      // TODO: remove this when API provides it!
+      // load receipt
+      if (item.receipt) item.documents.push(item.receipt)
+    }
+  }
 
-            // eslint-disable-next-line no-console
-            console.log(`Correction Creation error = ${error}`)
-            this.loadCorrectionDialog = true
-          }
-          break
-
-        case FilingTypes.CORRECTION:
-          // FUTURE: allow a correction to a correction?
-          // this.$router.push({ name: Routes.CORRECTION,
-          //   params: { correctedFilingId: item.filingId } })
-          alert('At this time, you cannot correct a correction. Please contact Ops if needed.')
-          break
-
-        case FilingTypes.ALTERATION:
-          // FUTURE: allow a correction to an alteration?
-          // this.$router.push({ name: Routes.CORRECTION,
-          //   params: { correctedFilingId: item.filingId } })
-          alert('At this time, you cannot correct an alteration. Please contact Ops if needed.')
-          break
-
-        default:
-          // fallback for all other filings
-          this.$router.push({
-            name: Routes.CORRECTION,
-            params: { correctedFilingId: item.filingId }
-          })
-          break
-      }
-    },
-
-    async downloadDocument (document: any, index: number) {
-      this.loadingDocument = true
-      this.downloadingDocIndex = index
-      await this.downloadOneDocument(document)
-      this.loadingDocument = false
-      this.downloadingDocIndex = -1
-    },
-
-    async downloadOneDocument (document: any) {
-      // safety check
-      if (!document.filingId || !document.filename) return
-
-      let url = `businesses/${this.entityIncNo}/filings/${document.filingId}`
-      const headers = { 'Accept': 'application/pdf' }
-
-      // Notice of articles or certificate will come in as a document report type
-      if (document.reportType) {
-        url += `?type=${document.reportType}`
-      }
+  /** Whether to disable correction for this history item. */
+  private disableCorrection (item: HistoryItemIF): boolean {
+    return (
+      this.disableChanges ||
+      item.availableOnPaperOnly ||
```

Comment:
Paper filings now cannot be corrected. (Previously they could but Colin filings couldn't.... and now these are merged.) This is still an outstanding question for Linda in the ticket description and I will update this if needed.